File: apps/flutter-app/lib/features/auth/providers/auth_state.dart
import '../../../core/models/user_model.dart';

class AuthState {
  const AuthState({
    this.user,
    this.isLoading = false,
    this.error,
  });

  final UserModel? user;
  final bool isLoading;
  final String? error;

  bool get isAuthenticated => user != null;

  AuthState copyWith({
    UserModel? user,
    bool? isLoading,
    String? error,
    bool clearUser = false,
    bool clearError = false,
  }) {
    return AuthState(
      user: clearUser ? null : user ?? this.user,
      isLoading: isLoading ?? this.isLoading,
      error: clearError ? null : error ?? this.error,
    );
  }
}

File: apps/angular-app/src/app/features/auth/store/auth.state.ts
import type { User } from '../../../core/models/user.model';

export type AuthState = {
  user: User | null;
  loading: boolean;
  error: string | null;
};

export const AUTH_INITIAL_STATE: AuthState = {
  user: null,
  loading: true,
  error: null,
};
